sdk/purview/Azure.Analytics.Purview.Scanning/src/Generated/PurviewScanClient.cs

```diff
@@ -76,6 +76,21 @@ public PurviewScanClient(Uri endpoint, string dataSourceName, string scanName, T
 
         /// <summary> Get a filter. </summary>
         /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <example>
+        /// This sample shows how to call GetFilterAsync and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
+        /// var endpoint = new Uri("<https://my-account-name.azure.com>");
+        /// var client = new PurviewScanClient(endpoint, credential);
```

Comment:
The root cause is that in the original codes, the sub-client is created using custom codes, not the recommended way (e.g., `CodeGenClient` attribute) in the doc. So, the code model doesn't contain metadata of sub-clients and corresponding factory methods. That's why the initialization codes are wrong here (and below).

I've fixed by using the `CodeGenClient` attribute. The side effect is that `PurviewScanClient.cs` is renamed to `PurviewScan`. In the `PureviewScan`, the example codes have the correct initialization logic.

Reply:
`PurviewScan` is Resource Client, so, according to the [guidelines for subclients](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclient-factory-methods-suffix), it is correct that it does not have the `Client` suffix.

Reply:
@annelo-msft Actually `PurviewScan` is **not** a Resource Client. See the [swagger definition](https://github.com/Azure/azure-rest-api-specs/blob/1c7df99f6a84335cfd7bf5be8c800d72c1dddbc2/specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2018-12-01-preview/scanningService.json#L774). It doesn't have a resource identifier parameter, which is the condition of a Resource Client, according to the generator code [IsResourceClient](https://github.com/archerzz/autorest.csharp/blob/00d2093b0d2d594fc23fdab7d63f1fb8f31cf81a/src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs#L64).
In the test project `ResourceClients-LowLevel` of `autorest.csharp`, there is [a resource identifier parameter in the swagger](https://github.com/archerzz/autorest.csharp/blob/78e59af2f193d7b445a21e054702a9416fafa341/test/TestProjects/ResourceClients-LowLevel/ResourceClients-LowLevel.json#L138), so the corresponding client is considered as a Resource Client.

The reason why it doesn't have `Client` suffix is because it's a SubClient. A SubClient doesn't have `Client` suffix, according to the generator code [here](https://github.com/archerzz/autorest.csharp/blob/b20e5ee38dce9d743b5c3390118433e2499ff4c1/src/AutoRest.CSharp/LowLevel/Output/LowLevelOutputLibraryFactory.cs#L88).